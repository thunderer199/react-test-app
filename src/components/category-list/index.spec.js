import React from 'react';

import renderer from 'react-test-renderer'
import {CategoryList} from "./index";
import {shallow} from "enzyme";

jest.mock('material-ui', () => ({
    List: ({children}) => <ul>{children}</ul>,
    ListItem: ({primaryText, onClick}) => <li onClick={onClick}>{primaryText}</li>,
    Divider: () => <hr/>
}), {virtual: true});

describe('testing category list component', () => {
    const noop = () => {};

    const fakeCategories = [
        {
            "name": "Электробезопасность",
            "alias": "electricity"
        },
        {
            "name": "Тендеры",
            "alias": "tender"
        }
    ];


    it('should rendering spinner when loading', () => {
        const snapshot = renderer.create(
            <CategoryList isLoading={true} getCategories={noop}/>
        ).toJSON();

        expect(snapshot).toMatchSnapshot();
    });

    it('should rendering with empty list', () => {
        const snapshot = renderer.create(
            <CategoryList categories={[]} getCategories={noop}/>
        ).toJSON();

        expect(snapshot).toMatchSnapshot();
    });

    it('should rendering with list', () => {
        const snapshot = renderer.create(
            <CategoryList categories={fakeCategories} getCategories={noop}/>
        ).toJSON();

        expect(snapshot).toMatchSnapshot();
    });

    it('should trigger get categories action', () => {
        const fn = jest.fn();
        shallow(
            <CategoryList categories={fakeCategories} getCategories={fn}/>
        );

        expect(fn).toHaveBeenCalledTimes(1);
    });

    it('should trigger select category action', () => {
        const fn = jest.fn();
        const element = shallow(
            <CategoryList categories={fakeCategories} getCategories={noop} selectCategory={fn}/>
        );

        const elem = element.find('List').first().childAt(0).find('ListItem');
        elem.simulate('click');

        expect(fn).toHaveBeenCalledWith({...fakeCategories[0]});
    });
});