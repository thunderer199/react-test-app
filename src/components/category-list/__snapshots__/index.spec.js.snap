// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing category list component should rendering spinner when loading 1`] = `
<div
  className="loader-wrapper"
>
  <div
    className="loader"
  />
</div>
`;

exports[`testing category list component should rendering with empty list 1`] = `
<div>
  <ul />
</div>
`;

exports[`testing category list component should rendering with list 1`] = `
<div>
  <ul>
    <div>
      <li
        onClick={[Function]}
      >
        Электробезопасность
      </li>
      <hr />
    </div>
    <div>
      <li
        onClick={[Function]}
      >
        Тендеры
      </li>
      <hr />
    </div>
  </ul>
</div>
`;
